india_compliance/gst_india/utils/e_invoice.py

```diff
@@ -108,6 +128,24 @@ def generate_e_invoice(docname, throw=True):
             # IRN details cannot be provided as it is generated more than 2 days ago
             result = result.Desc if response.error_code == "2283" else response
 
+    except GatewayTimeoutError:
+        doc.db_set({"einvoice_status": "Auto-Retry"})
+        e_invoice_auto_retry_after = now_datetime() + timedelta(minutes=5)
+
+        frappe.db.set_single_value(
+            "GST Settings", "e_invoice_auto_retry_after", e_invoice_auto_retry_after
+        )
+
+        frappe.msgprint(
+            _(
+                "Government services are currently experiencing downtime, resulting in high response times and a Gateway Timeout error. We apologize for the inconvenience caused. Your e-invoice generation will be automatically retried after {0} minutes."
+            ).format(str(e_invoice_auto_retry_after.replace(microsecond=0))),
+            _("Warning"),
+            indicator="yellow",
+        )
+
+        return
```

Comment:
This should be `return False` instead. And then that should be captured by generate_e_invoices to prevent further execution. e-Invoice status for pending invoices should ideally be set in one query itself.